src/lib/config.ts

```diff
@@ -1235,15 +1230,7 @@ export async function getAsync(
 ) {
     if (INITIALISED) {
         // TODO: consider storing keys directly
-        let browserconfig
-        switch (get("storageloc")) {
-            case "local":
-                browserconfig = await browser.storage.local.get(CONFIGNAME)
-                break
-            case "sync":
-                browserconfig = await browser.storage.sync.get(CONFIGNAME)
-                break
-        }
+        const browserconfig = await browser.storage.local.get(CONFIGNAME)
         USERCONFIG = browserconfig[CONFIGNAME] || o({})
```

Comment:
[browser.storage.X.get](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/get) has a mechanism to get it to return a default object. We should consider using that someday.